src/MessageEditor.js

```diff
@@ -14,23 +21,69 @@ class MessageEditor extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (this.props.id !== prevProps.id && this.props.id) {
+      this.setState({
+        messages: [],
+        gauges: []
+      });
       api.fetch(`glyphs/${this.props.id}/messages`)
       .then(messages => this.setState({ messages: messages }));
 
       api.fetch(`glyphs/${this.props.id}/gauges`)
       .then(gauges => this.setState({ gauges: gauges }));
+
     }
   }
 
   render() {
-    const { messages, gauges } = this.state;
+    let { messages, gauges } = this.state;
+
+    const addVMessage = () => {
+      api.post(`glyphs/${this.props.id}/messages/`, {"pass" : this.inputRef.value, "text" : 'Default message', "probability" : 0}).then(result => {console.log(result)})
+      messages.push({"text" : 'Default message', "probability" : 0})
```

Comment:
Shouldn't this line be inside the continuation (`.then`, I am not sure if this is the right vocabulary) of the above?  That way, we can make sure to only add the message if there is not an error.  Maybe I'll find the code elsewhere.